src/lib/notion/events.ts

```diff
@@ -15,23 +16,10 @@ export type EventDataType = {
 
 export const EVENTS_DATABASE_ID = 'f988151abd6448ebb70053c5ca1278f9';
 
-export async function getEvent(id: string) {
-  const page = (await notion.pages.retrieve({ page_id: id })) as any;
-  let date = '';
-  if (page.properties.Date.date.start.includes('T')) {
-    date = format(parseISO(page.properties.Date.date.start), 'MMMM dd, yyyy h:mm a');
-  } else {
-    date = format(parseISO(page.properties.Date.date.start), 'MMMM dd, yyyy');
-  }
-  const event: EventDataType = {
-    eventPageId: id,
-    eventName: page.properties.Name.title[0].text.content,
-    date: date,
-    venue: page.properties.Venue.rich_text[0].plain_text,
-    status: page.properties.Status.status.name,
-    description: page.properties.Description.rich_text[0]?.plain_text || '',
-    coverURL: page.properties['Cover URL'].rich_text[0]?.plain_text || '/default',
-  };
+export async function getEvent(slug: string) {
+  const events = await getEvents();
+  const event = events.find(e => e.slug === slug);
+  if (!event) throw new Error(`Event with slug ${slug} not found`);
```

Comment:
Do you think we can handle this error more gracefully? Might be time to set up a not found page so that we don't get the default Next.js **500 | Internal Server Error page**

<img width="1494" alt="image" src="https://github.com/user-attachments/assets/4b984473-0991-4297-8abd-1d64e880bbea">

Reply:
Yes, we need a global 404 page. I can create an issue for this. It won't be a part of this PR

Reply:
Created #143 and assigned it to myself

Reply:
Read what you said more carefully. I'll update the events and project pages so instead of throwing errors they fall back to a not found page. #143 will exist to implement this for the rest of the site